### PR TITLE
Dev server file watchers

### PR DESCRIPTION
**Pull request by SamTolmay**
Closes ~#ISSUE_NUMBER~

### What are the changes and their implications?

This pull request adds the following:
- Fixes dev server 404 redirect.
- Modifies readFile not throw if path is not resolved.
- Reads the env vars in .env file, and passes to dev server.
- Updates dev server if env vars change.
- Exits dev server if config Lowdefy version changes.

## Checklist

- [x] Pull request is made to the "develop" branch
- [ ] Tests added
- [ ] Documentation added/updated
- [x] Code has been formatted with Prettier
- [x] Edits from maintainers are allowed

### CODE REVIEW

File: packages/server-dev/src/manager/processes/restartServer.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import startServerProcess from './startServerProcess.mjs';
+
+function shutdownServer(context) {
```
Comment (Gervwyk) [inline packages/server-dev/src/manager/processes/restartServer.mjs:19]: rename?

File: packages/server-dev/src/manager/getContext.mjs [hunk 2]
```diff
@@ -38,25 +42,19 @@ async function getContext() {
...
-      if (context.serverProcess) {
-        console.log('Shutting down server...');
-        context.serverProcess.kill();
-      }
-    },
     verbose,
   };
+
+  context.version = process.env.npm_package_version;
+  context.initialBuild = initialBuild(context);
```
Comment (Gervwyk) [inline packages/server-dev/src/manager/getContext.mjs:49]: Doesn't initialBuild do the next four steps aswell?

File: packages/server-dev/src/manager/getContext.mjs [hunk 3]
```diff
@@ -38,25 +42,19 @@ async function getContext() {
...
   context.installPlugins = installPlugins(context);
   context.lowdefyBuild = lowdefyBuild(context);
   context.nextBuild = nextBuild(context);
+  context.readDotEnv = readDotEnv(context);
   context.reloadClients = reloadClients(context);
+  context.restartServer = restartServer(context);
+  context.shutdownServer = shutdownServer(context);
```
Comment (Gervwyk) [inline packages/server-dev/src/manager/getContext.mjs:56]: restart then shut down?

File: packages/server-dev/src/manager/processes/startWatchers.mjs [hunk 4]
```diff
@@ -57,8 +55,10 @@ Watch <server-dir>/build/plugins/*
 - Restart server.
 */
 
-async function startWatchers(context) {
-  await Promise.all([configWatcher(context), envWatcher(context)]);
+function startWatchers(context) {
+  return async () => {
```
Comment (Gervwyk) [inline packages/server-dev/src/manager/processes/startWatchers.mjs:59]: If you are returning a method, maybe getStartWatchers is a better name? Not sure how it is used.

File: packages/server-dev/src/manager/watchers/configWatcher.mjs [hunk 5]
```diff
@@ -13,11 +13,24 @@
...
-import setupWatcher from './setupWatcher.mjs';
+import getLowdefyVersion from '../utils/getLowdefyVersion.mjs';
+import setupWatcher from '../utils/setupWatcher.mjs';
 
 async function configWatcher(context) {
-  const callback = async () => {
+  const callback = async (filePaths) => {
+    const lowdefyYamlModified = filePaths
+      .flat()
+      .some((filePath) => filePath.includes('lowdefy.yaml') || filePath.includes('lowdefy.yml'));
```
Comment (Gervwyk) [inline packages/server-dev/src/manager/watchers/configWatcher.mjs:25]: lowdefy.json

Comment (SamTolmay) [reply packages/server-dev/src/manager/watchers/configWatcher.mjs:25]: We don't support lowdefy.json in the CLI. I guess we should?

**Review by Gervwyk — APPROVED** [review]

File: packages/server-dev/src/manager/watchers/configWatcher.mjs (hunk 5 above)
Comment (Gervwyk) [reply packages/server-dev/src/manager/watchers/configWatcher.mjs:25]: Would be nice.

**Pull request merged**